### PR TITLE
Metrics Rework

### PR DESCRIPTION
**Pull request by GalaxyGorilla**
The former metrics implementation had some issues:

  - handle time and counter metrics were not complete
  - it was hard/impossible to identify server and their nases based on the exometer id
  - no client tags (see #59)
  - no concise exometer id structure
  - subscription options were tied to the influxdb reporter
   
I (hopefully) solved these issues. An exometer id structure now looks like this:

[eradius, radius, EventName, EventType, Role,  LocationID, LocationIP, LocationPort, Peer, PeerIP, PeerPort, Unit]

For example:

[eradius, radius, request, accounting, server, ServerName, ServerIP, ServerPort, NasName, NasIP, undefined, ticks] 

Note: The NAS port is undefined because it is random.

In a nutshell one can now expose metrics that are approximately compliant with RFC 2619, 2621, 4672 and 2618. One big drawback at the moment is that error situations cannot be associated with request types. Example: Only retransmissions are counted and NOT retransmissions per access request, accounting request, etc. Also there is no association with attribute content which is important for CoA clients in RFC 4672.

### CODE REVIEW

**Comment by GalaxyGorilla** [comment]
This is not tested. I will update this PR accordingly.

**Comment by GalaxyGorilla** [comment]
Need to update tests and some performance tweaking.

**Comment by GalaxyGorilla** [comment]
This is ready to be merged.

**Comment by RoadRunnr** [comment]
This is nice example of how a pull request should NOT look like. Lumping a large rewrite into in big commit makes I difficult to properly review the changes.

A large rewrite should be broken into a series of small, incremental changes. Each change should be consistent in itself.

A good reference is  https://www.kernel.org/doc/Documentation/SubmittingPatches, especially chapter 3.

Do't waste time on redoing this pull request, but use a different approach in the future.

**Comment by GalaxyGorilla** [comment]
@RoadRunnr Totally agree. Unfortunately I didn't even know eradius well before I started, that's why it got a bit messy in the working process ...

**Comment by GalaxyGorilla** [comment]
The last update is about pending server request (e.g. pending nas callbacks). Before this was only available for clients.

**Comment by GalaxyGorilla** [comment]
It is left to chose appropiate histogram intervals / time frames. Hence the WiP tag.

**Comment by GalaxyGorilla** [comment]
Histogram time spans set to 60s.
